common/primitives/src/messages.rs

```diff
@@ -15,9 +15,9 @@ use utils::*;
 pub struct MessageResponse<AccountId, BlockNumber> {
 	#[cfg_attr(feature = "std", serde(with = "as_hex"))]
 	pub data: Vec<u8>, //  Serialized data in a user-defined schema format
-	pub signer: AccountId,       //  Signature of the signer
-	pub msa_id: MessageSenderId, //  Message source account id (the original sender)
-	pub index: u16,              // index in block to get total order
+	pub signer: AccountId,              //  Signature of the signer
+	pub sender_msa_id: MessageSenderId, //  Message source account id (the original sender)
```

Comment:
I think sender might still cause confusion, how about `origin_msa_id`?

Reply:
lol, I think origin might also be confusing since in extrinsic we have `origin` caller which is our case can be provider